### PR TITLE
Adds support to make db-connection timeout configurable via CLI.

### PR DESCRIPTION
**Pull request by Harkishen-Singh**
Signed-off-by: Harkishen-Singh <harkishensingh@hotmail.com>

### CODE REVIEW

File: pkg/util/lock.go [hunk 1]
```diff
@@ -202,7 +200,7 @@ func (l *PgAdvisoryLock) getConn(connStr string, cur, maxRetries int) (*pgx.Conn
 		return nil, fmt.Errorf("max attempts reached. giving up on getting a db connection")
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), waitForConnectionTimeout)
+	ctx, cancel := context.WithTimeout(context.Background(), l.connTimeout)
```
Comment (cevian) [inline pkg/util/lock.go:203]: I think the code should only impose a timeout if one isn't defined by the connection string. Remember a connection string can be define by the -db-* parameters (including -db-connection-timeout) OR by the -db-uri 
parameter. We should only impose a timeout here if the connection string doesn't have one. We can use something like:

```
      ctx := context.Background()
      cfg, err := pgx.ParseConfig(connStr)
      if cfg.ConnectTimeout == 0 {
            ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
            defer cancel()
      }

```

**Review by cevian — CHANGES_REQUESTED** [review]

File: pkg/util/lock.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Harkishen-Singh) [reply pkg/util/lock.go:203]: This has been changed in the recent commits, so resolving this.

File: pkg/util/lock.go [hunk 2]
```diff
@@ -202,14 +200,22 @@ func (l *PgAdvisoryLock) getConn(connStr string, cur, maxRetries int) (*pgx.Conn
...
-	defer cancel()
-
 	cfg, err := pgx.ParseConfig(connStr)
 	if err != nil {
 		return nil, fmt.Errorf("error parsing config connection: %w", err)
 	}
 
+	var (
+		ctx    = context.Background()
+		cancel context.CancelFunc
```
Comment (cevian) [inline pkg/util/lock.go:210]: nit: why declare this outside the if statement, it's only used inside. Just use the `:=` operator to create the cancel inside the if.  

The reason the current code is worse is that currently cancel survive to after line 217 but would be uninitiated if the if block isn't hit. Uninitiated variables are bad and should be avoided if possible.

**Review by cevian — APPROVED** [review]
One code quality issue to fix but otherwise looks good.

File: pkg/util/lock.go (hunk 2 above)
Comment (Harkishen-Singh) [reply pkg/util/lock.go:210]: yep, that's my mistake. I thought it's being used somewhere else which is not local to that block. Anyways, thanks for pointing it out.

Comment (Harkishen-Singh) [reply pkg/util/lock.go:210]: Well, the reason is if i do not declare it here then the `:=` operator will make ctx too local in the block which is a compilation error. So, in order to allow ctx accessible to the below block and that goes into the ConnectConfig, I need to declare both of them outside.

> The reason the current code is worse is that currently cancel survive to after line 217 but would be uninitiated if the if block isn't hit.

If I am not wrong, I read somewhere that the difference between initializing `:=` vs `var` is that the memory is immediately used in `:=` but in `var`, its not used unless the variable is used somewhere. Do you think this is wrong? @cevian

Comment (Harkishen-Singh) [reply pkg/util/lock.go:210]: but still, it looks weird. I agree. I have a workaround that fits you suggestion. Will implement that and push. Thank you.

**Review by VineethReddy02 — APPROVED** [review]

**Pull request merged**